lib/Dialect/Ibis/IbisOps.cpp

```diff
@@ -20,6 +20,18 @@ using namespace mlir;
 using namespace circt;
 using namespace ibis;
 
+// Looks up a `sym`-symbol defining operation of type T in the `mlir::ModuleOp`
+// parent scope of the provided `base` operation.
+template <typename T>
+static T lookupInModule(Operation *base, FlatSymbolRefAttr sym,
+                        SymbolTable *symbolTable) {
+  auto mod = base->getParentOfType<mlir::ModuleOp>();
+  if (symbolTable)
+    return dyn_cast<T>(symbolTable->lookupSymbolIn(mod, sym));
+
+  return mod.lookupSymbol<T>(sym);
```

Comment:
I'm not wild about these functions which change algorithmic performance based on the presence of an argument. Too easy to abuse, too hard to detect visually.

Reply:
It is being used transitively throuh i.e. `InstanceOp::getReferencedModuleSlow`.